### PR TITLE
Mix bugfixes

### PR DESCRIPTION
**Pull request by arkpar**
* Fixed issues having multiple contracts in the same source
* Fixed issues navigating between pages in web preview
* #1714
* Added tests

### CODE REVIEW

File: mix/CodeModel.cpp [hunk 1]
```diff
@@ -279,9 +279,22 @@ void CodeModel::runCompilationJob(int _jobId)
 				QQmlEngine::setObjectOwnership(contract, QQmlEngine::CppOwnership);
```
Comment (chriseth) [inline mix/CodeModel.cpp]: Quite long function. Does it make sense to create a new function where the guards start their work?

File: mix/HttpServer.cpp [hunk 2]
```diff
@@ -145,16 +145,23 @@ void HttpServer::readClient()
...
 			{
 				QUrl url(hdr.split(' ')[1]);
 				QString l;
 				do
+				{
 					l = socket->readLine();
+					//collect headers
+					int colon = l.indexOf(':');
+					if (colon > 0)
+						headers[l.left(colon)] = l.right(l.length() - colon - 1);
```
Comment (chriseth) [inline mix/HttpServer.cpp]: Perhaps lowercase the key?

File: mix/qml/WebPreview.qml [hunk 3]
```diff
@@ -184,8 +185,10 @@ Item {
 						content = fileIo.readFile(doc.path);
 				}
 
-				if (documentName === urlInput.text.replace(httpServer.url + "/", "")) {
-					//root page, inject deployment script
+				var accept = _request.headers["Accept"];
+				if (accept && accept.indexOf("text/html") >= 0 && !_request.headers["HTTP_X_REQUESTED_WITH"])
```
Comment (chriseth) [inline mix/qml/WebPreview.qml]: That looks quite hackish. Which problem does it solve?

File: mix/HttpServer.cpp (hunk 2 above)
Comment (arkpar) [inline mix/HttpServer.cpp]: HTTP headers are case sensitive

File: mix/qml/WebPreview.qml (hunk 3 above)
Comment (arkpar) [inline mix/qml/WebPreview.qml]: This is do differentiate ajax requests from navigation requests. We only need to inject web3 object if the page request is coming from the browser, and not from javascript

File: mix/HttpServer.cpp (hunk 2 above)
Comment (chriseth) [inline mix/HttpServer.cpp]: To my knowledge, they are not: https://tools.ietf.org/html/rfc7230#section-3.2

Comment (arkpar) [inline mix/HttpServer.cpp]: You're absolutely right

**Pull request merged**